### PR TITLE
chore: Splitting JS Object body dump vs parsed JS Object update to remove race condition originated bad state

### PR DESCRIPTION
**Pull request by trishaanand**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9029211697>
> Commit: 190f7682cad3117e776d94f1502609026633369c
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9029211697&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/ActionCollectionControllerCE.java [hunk 1]
```diff
@@ -130,6 +130,18 @@ public Mono<ResponseDTO<ActionCollectionDTO>> updateActionCollection(
                 .map(updatedResource -> new ResponseDTO<>(HttpStatus.OK.value(), updatedResource, null));
     }
 
+    @JsonView(Views.Public.class)
+    @PutMapping("/{id}/body")
+    public Mono<ResponseDTO<Integer>> updateActionCollectionBody(
+            @PathVariable String id,
+            @Valid @RequestBody String body,
+            @RequestHeader(name = FieldName.BRANCH_NAME, required = false) String branchName) {
+        log.debug("Going to update action collection body with id: {}, branch: {}", id, branchName);
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/ce/ActionCollectionControllerCE.java:139]: Nit: Do we need these kinda log lines? I think we should start getting rid of them.

**Review by nidhi-nair — APPROVED** [review]

**Comment by subrata71** [comment]
QQ: Doesn't this PR need the client-side changes w.r.t. modules and workflows? @trishaanand Asking this because there is JS object in both workflows and modules.

**Comment by trishaanand** [comment]
> QQ: Doesn't this PR need the client-side changes w.r.t. modules and workflows? @trishaanand Asking this because there is JS object in both workflows and modules.

@ApekshaBhosale : Is this correct? Do we need changes in modules and workflows code flow as well?

**Comment by ApekshaBhosale** [comment]
@trishaanand i don't think so because underline uses same JS objects. @ashit-rath what about workflows?

**Comment by ApekshaBhosale** [comment]
@trishaanand checked code for workflow as well. looks like it uses same underline js object so it will be fine!

**Comment by ApekshaBhosale** [comment]
@dvj1988 @trishaanand format of JS object body is different so regex is failing on client side. extra "/" i see

**Review by dvj1988 — APPROVED** [review]

**Pull request merged**